src/Parallel/NodeLock.hpp

```diff
@@ -8,66 +8,93 @@
 #include "Utilities/Gsl.hpp"
 #include "Utilities/NoSuchType.hpp"
 
+#include "Parallel/Printf.hpp"
+
 namespace Parallel {
 /*!
  * \ingroup ParallelGroup
- * \brief Create a converse CmiNodeLock
+ * \brief A typesafe wrapper for the Converse nodelock, with safe creation,
+ * destruction, and serialization.
+ *
+ * \details On construction, this class creates a Converse nodelock, and frees
+ * the lock on destruction. During serialization, the contained nodelock at the
+ * sending node is destroyed to avoid misuse, and throws an error if the lock is
+ * used thereafter. When the lock is deserialized, the nodelock is re-created.
  */
-inline CmiNodeLock create_lock() noexcept { return CmiCreateLock(); }
+class NodeLock {
+ public:
+  NodeLock() noexcept {
```

Comment:
Could we move the definitions to a `cpp`? I realize it won't make a huge compilation time difference, but every bit helps :)